app/Platform/Actions/UpdateGameAchievementsMetricsAction.php

```diff
@@ -82,4 +134,78 @@ public function execute(Game $game): void
 
         // TODO GameAchievementSetMetricsUpdated::dispatch($game);
     }
+
+    private function batchUpdateAchievements(array $updates): void
+    {
+        if (empty($updates)) {
+            return;
+        }
+
+        /**
+         * Eloquent does not allow us to use `upsert()` for this, as `upsert()` requires
+         * every single field in the model to be passed in. This uses too much memory.
+         * Performance matters here - we'll use the DB facade and raw SQL instead.
+         *
+         * UPDATE Achievements
+         * SET
+         *   unlocks_total = CASE ID
+         *     WHEN X THEN Y
+         *   END,
+         *   unlocks_hardcore_total = CASE ID
+         *     WHEN X THEN Y
+         *   END,
+         *   unlock_percentage = CASE ID
+         *     WHEN X THEN Y
+         *   END,
+         *   unlock_hardcore_percentage = CASE ID
+         *     WHEN X THEN Y
+         *   END,
+         *   TrueRatio = CASE ID
+         *     WHEN X THEN Y
+         *   END,
+         *   Updated = "..."
+         * WHERE
+         *   ID IN ( ... );
+         */
```

Comment:
I don't see how this could possibly be faster than doing a `save()` on each modified record. The query engine will have to process each case statement for every row.